modules/core/src/lib/layer.js

```diff
@@ -288,6 +288,14 @@ export default class Layer extends Component {
     return index;
   }
 
+  // Return the closest pickable and visible object at the given screen coordinate.
+  pickObject({x, y, radius = 0}) {
+    if (this.context.deck) {
+      return this.context.deck.pickObject({x, y, radius, layerIds: [this.props.id]});
```

Comment:
I want to be able to specify `layerIds` in case of composite layer.

Also want the multi object method.

Reply:
> I want to be able to specify layerIds in case of composite layer.

I think you get them all. As I recall this matches all layers with that prefix, which per convention includes all sub layers of composite layers (their ids are appended to the parent). Excluding some of them is not possible with this mechanism though.

Reply:
But I might not want all of them.

Reply:
Good point. Ideally you should be able to inspect the pick infos after picking is done, but I am not 100% sure we save the id of the origin layer.

But seems weird to open up layerIds in Layer which would allow picking other layers.

Reply:
Weird yes, but I don't think there are any security issues.

Reply:
If we want the same semantics as the `Deck` method, why don't we just make sure the `Deck` instance is available in the `Layer.context` instead of adding more methods, test surface, bundle size etc.

Reply:
> If we want the same semantics as the `Deck` method, why don't we just make sure the `Deck` instance is available in the `Layer.context` instead of adding more methods, test surface, bundle size etc.

That is what I asked for in the first place 🎉

Reply:
The deck instance is still there, you can use `this.context.deck.pickObject()` as the old way `this.context.layerManager.pickObject()` if there is special requirement.

Reply:
Alternatively we can accept a list of sublayer ids (i.e. layer ids minus the parent layer id, for example http://deck.gl/#/documentation/deckgl-api-reference/layers/geojson-layer?section=sub-layers)

Reply:
> Alternatively we can accept a list of sublayer ids (i.e. layer ids minus the parent layer id, for example http://deck.gl/#/documentation/deckgl-api-reference/layers/geojson-layer?section=sub-layers)

we can add another pickObject API in composite layer class with extra subLayersId prop

Reply:
Just another 2 cents:

If the worry is that `this.context.deck.pickObject()` is too ugly/low level, how about just adding an accessor for `this.context.deck`?

```
class Layer {
  get deck() { return this.context.deck; }
  onClick() {
    this.deck.pickObject(...);
  }
}
```

This way, there are no new methods to document, test, and maintain.

Then we make any enhancements to layer filtering in the global deck method so as to benefit all use cases.

Reply:
> Just another 2 cents:
> 
> If the worry is that `this.context.deck.pickObject()` is too ugly/low level, how about just adding an accessor for `this.context.deck`?
> 
> ```
> class Layer {
>   get deck() { return this.context.deck; }
>   onClick() {
>     this.deck.pickObject(...);
>   }
> }
> ```
> This way, there are no new methods to document, test, and maintain.
> 
> Then we make any enhancements to layer filtering in the global deck method so as to benefit all use cases.

Agree 100% 💯